core/src/main/scala/org/typelevel/paiges/Doc.scala

```diff
@@ -567,11 +567,12 @@ object Doc {
    */
   private[paiges] case class Align(doc: Doc) extends Doc
 
-  /**
-   * Invariant: generate is referentially transparent
-   */
-  private[paiges] case class LazyDoc(private val generate: () => Doc) extends Doc {
-    lazy val evaluated: Doc = generate()
+  private[paiges] case class LazyDoc(thunk: LazyDoc.Thunk) extends Doc
+
+  private[paiges] object LazyDoc {
+    class Thunk(value: => Doc) {
```

Comment:
can we make this `extends Serializable`

Reply:
Sure.  Do we serialize these?  I thought we only use it to avoid warnings about inferring products and such due to `case class` code generation mucking with type inference.

Reply:
We don't but users might with akka, spark, scalding, etc...

Reply:
Should we be testing serialization?  I don't know how it works in Java/Scala, but I don't see how it can work with function arguments and lazy values.